nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -855,6 +867,21 @@ in
                   The address must be in the default VLAN (10.0.2.0/24).
               '';
           }
+          { assertion = cfg.useDefaultFilesystems -> cfg.useDefaultPartitionTable;
+            message =
+              ''
+                You cannot use a non-default partition table with the default filesystems.
+              '';
+          }
+          {
+            assertion = !cfg.useDefaultPartitionTable -> !cfg.useEFIBoot;
```

Comment:
This is another reason why we shouldn't (always) read the efivars from the disk image or why they shouldn't even be created in the disk image.

Reply:
You mean to separate EFIVARS and disk images? I mean, I would agree, but what is the alternative?